pkg/router/gloo.go

```diff
@@ -39,15 +39,23 @@ type GlooRouter struct {
 	glooClient          clientset.Interface
 	flaggerClient       clientset.Interface
 	logger              *zap.SugaredLogger
-	upstreamDiscoveryNs string
 }
 
 // Reconcile creates or updates the Gloo Edge route table
 func (gr *GlooRouter) Reconcile(canary *flaggerv1.Canary) error {
-	apexName, _, _ := canary.GetServiceNames()
-	canaryName := fmt.Sprintf("%s-%s-canary-%v", canary.Namespace, apexName, canary.Spec.Service.Port)
-	primaryName := fmt.Sprintf("%s-%s-primary-%v", canary.Namespace, apexName, canary.Spec.Service.Port)
-
+	apexName, primaryName, canaryName := canary.GetServiceNames()
+	canaryUpstreamName := fmt.Sprintf("%s-%s-canaryUpstream-%v", canary.Namespace, apexName, canary.Spec.Service.Port)
+	primaryUpstreamName := fmt.Sprintf("%s-%s-primaryUpstream-%v", canary.Namespace, apexName, canary.Spec.Service.Port)
+	canaryUs := getGlooUpstreamForKubeService(canary, canaryUpstreamName, canaryName)
```

Comment:
let's add a comment here explaining why we do this instead of using kube service destination or relying on discovery, for the next reader